v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,292 @@
+    [ ca ]
+    default_ca = CA_default
+
+    [ CA_default ]
+    default_days = 3660
+    database = index.txt
+    serial = serial.txt 
+    default_md = sha256 
+    copy_extensions = copy
+
+    # Used to generate the CA certificate.
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    x509_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = Cockroach CA
+
+    [ extensions ]
+    keyUsage = critical,digitalSignature,nonRepudiation,keyEncipherment,keyCertSign
+    basicConstraints = critical,CA:true,pathlen:1
+
+    # Common policy for nodes and users.
+    [ signing_policy ]
+    organizationName = supplied
+    commonName = supplied
+
+    # Used to sign node certificates.
+    [ signing_node_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = serverAuth,clientAuth
+
+    # Used to sign client certificates.
+    [ signing_client_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = clientAuth
+    ~~~
+
+3. Generate the CA key using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    openssl genrsa -out ${my-safe-directory}/ca.key 2048 2> /dev/null
+    chmod 400 ${my-safe-directory}/ca.key
+    ~~~
+
+4. Generate the CA certificate using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create CA certificate.
+    openssl req \
+      -new \
+      -x509 \
+      -config ca.conf \
+      -key ${CCERTS}/ca.key \
+      -out ${OSCERTS}/ca.crt \
+      -days 3660 \
+      -batch
+
+    # Reset database and index files.
+    rm -f index.txt serial.txt
+    touch index.txt
+    echo '01' > serial.txt
+    ~~~
+
+### Generate the certificate and key pairs for nodes
+
+1. Create the node.conf file for the first node:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    req_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = <node-hostname>
+
+    [ extensions ]
+    subjectAltName = DNS:<node-hostname>,DNS:<node-domain>,IP:<IP Address>
+    ~~~
+
+2. Generate the key for the first node using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    openssl genrsa -out ${certs}/node.key 2048 2> /dev/null
+    chmod 400 ${certs}/node.key
+    ~~~
+
+3. Generate the CSR for the first node using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create Node certificate signing request.
+    openssl req \
+      -new \
+      -config node.conf \
+      -key ${certs}/node.key \
+      -out ${certs}/node.csr \
```

Comment:
we want to store the csr somewhere else so it doesn't end up in the final `certs` directory. Could be `/tmp/` or even just the current directory (eg: `-out /tmp/node.csr` or `-out node.csr`).
Alternatively, we could delete it after generating the final `.crt`

Reply:
Can we leave out the directory altogether? I remember discussing that the working directory will be used in that case, and that will be okay.

Reply:
For CSR, leave out the directory